src/main/java/es/eucm/gleaner/realtime/states/MongoGameplayState.java

```diff
@@ -18,35 +18,82 @@
 import com.mongodb.BasicDBObject;
 import com.mongodb.DB;
 import com.mongodb.DBObject;
+import es.eucm.gleaner.realtime.elastic.EsConfig;
+import es.eucm.gleaner.realtime.elastic.StormElasticSearchClient;
+import es.eucm.gleaner.realtime.elastic.trident.EsState;
+import es.eucm.gleaner.realtime.topologies.KafkaTopology;
 import es.eucm.gleaner.realtime.utils.DBUtils;
 import org.bson.types.ObjectId;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.update.UpdateRequest;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import storm.trident.state.OpaqueValue;
 
+import java.io.IOException;
+import java.util.Date;
 import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import static org.elasticsearch.common.xcontent.XContentFactory.*;
 
 public class MongoGameplayState extends GameplayState {
 	private static final Logger LOG = LoggerFactory
 			.getLogger(GameplayState.class);
+	private final EsConfig esConfig;
 
 	private DB db;
 
-	public MongoGameplayState(DB db) {
+	public MongoGameplayState(DB db, EsConfig esConfig) {
 		this.db = db;
+		this.esConfig = esConfig;
+	}
+
+	public void prepare() {
+		if (esConfig == null) {
+			return;
+		}
+		try {
+			synchronized (EsState.class) {
+				if (EsState.client == null) {
+					EsState.client = new StormElasticSearchClient(esConfig)
```

Comment:
The MongoGamePlayState with an instance of EsState??? Should exist a ElasticSearchGameplayState
and split the TridentStream as we spoke